Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,744 @@
+        pol_directions = set()
+        numors = set()
+        for name in mtd[ws].getNames():
+            last_underscore = name.rfind("_")
+            numors.add(name[:last_underscore])
+            pol_directions.add(name[last_underscore+1:])
+        if len(numors) > 1:
+            names_list = []
+            for direction in sorted(list(pol_directions)):
+                name = '{0}_{1}'.format(ws, direction)
+                list_pol = []
+                for numor in numors:
+                    if average_detectors:
+                        try:
+                            Plus(LHSWorkspace=name, RHSWorkspace=mtd[numor + '_' + direction], OutputWorkspace=name)
+                        except ValueError:
+                            CloneWorkspace(InputWorkspace=mtd[numor + '_' + direction], OutputWorkspace=name)
+                    else:
+                        list_pol.append('{0}_{1}'.format(numor, direction))
+                if average_detectors:
+                    CreateSingleValuedWorkspace(DataValue=len(numors), OutputWorkspace='norm')
+                    Divide(LHSWorkspace=name, RHSWorkspace='norm', OutputWorkspace=name)
+                    DeleteWorkspace(Workspace='norm')
+                else:
+                    SumOverlappingTubes(','.join(list_pol), OutputWorkspace=name,
+                                        OutputType='1D', ScatteringAngleBinning=self.getProperty('ScatteringAngleBinSize').value,
+                                        Normalise=True, HeightAxis='-0.1,0.1')
+                names_list.append(name)
+            DeleteWorkspaces(WorkspaceList=ws)
+            GroupWorkspaces(InputWorkspaces=names_list, OutputWorkspace=ws)
+        return ws
+
+    def _subtract_background(self, ws, container_ws, transmission_ws):
+        """Subtracts empty container and absorber scaled by transmission."""
+        absorber_ws = self.getPropertyValue('AbsorberInputWorkspace')
+        if absorber_ws == "":
+            return ws
+        unit_ws = 'unit_ws'
+        CreateSingleValuedWorkspace(DataValue=1.0, OutputWorkspace=unit_ws)
+        background_ws = 'background_ws'
+        tmp_names = [unit_ws, background_ws]
+        nMeasurements = self._data_structure_helper()
+        singleContainerPerPOL = mtd[container_ws].getNumberOfEntries() < mtd[ws].getNumberOfEntries()
+        singleAbsorberPerPOL = mtd[container_ws].getNumberOfEntries() < mtd[ws].getNumberOfEntries()
+        for entry_no, entry in enumerate(mtd[ws]):
+            if singleContainerPerPOL:
+                container_entry = mtd[container_ws][entry_no % nMeasurements].name()
+            else:
+                container_entry = mtd[container_ws][entry_no].name()
+            if singleAbsorberPerPOL:
+                absorber_entry = mtd[absorber_ws][entry_no % nMeasurements].name()
+            else:
+                absorber_entry = mtd[absorber_ws][entry_no].name()
+            mtd[container_entry].setYUnit('')
```

Comment:
Why do you need these lines? Is it the loader that gives workspaces with inappropriate Y units, that you have to reset to empty here?

Reply:
Indeed, these lines are not required anymore. They were needed to enforce conformity of the `YUnit` to mitigate any issues in `Plus` or `Minus` algorithms.